backend/entities/application_entity.py

```diff
@@ -0,0 +1,386 @@
+        """
+
+        entity = super().from_model(model)
+        entity.academic_hours = model.academic_hours
+        entity.extracurriculars = model.extracurriculars
+        entity.expected_graduation = model.expected_graduation
+        entity.program_pursued = model.program_pursued
+        entity.other_programs = model.other_programs
+        entity.gpa = model.gpa
+        entity.comp_gpa = model.comp_gpa
+        entity.comp_227 = model.comp_227
+        entity.preferred_sections = [
+            SectionEntity.from_model(section) for section in model.preferred_sections
+        ]
+
+        return entity
+
+    def to_model(self) -> UTA:
+        """
+        Converts an `ApplicationEntity` object into an `Application` model object
+
+        Returns:
+            Application: `Application` object from the entity
+        """
+
+        parent_model = super().to_model().model_dump()
+        return UTA(
+            **parent_model,
+            academic_hours=self.academic_hours,
+            extracurriculars=self.extracurriculars,
+            expected_graduation=self.expected_graduation,
+            program_pursued=self.program_pursued,
+            other_programs=self.other_programs,
+            gpa=self.gpa,
+            comp_gpa=self.comp_gpa,
+            comp_227=self.comp_227,
+        )
+
+    def to_details_model(self) -> UTADetails:
+        """
+        Converts a `ApplicationEntity` object into a `ApplicationDetails` model object
+
+        Returns:
+            ApplicationDetails: `ApplicationDetails` object from the entity
+        """
+
+        parent_model = super().to_details_model().model_dump()
+
+        return UTADetails(
+            **parent_model,
+            academic_hours=self.academic_hours,
+            extracurriculars=self.extracurriculars,
+            expected_graduation=self.expected_graduation,
+            program_pursued=self.program_pursued,
+            other_programs=self.other_programs,
+            gpa=self.gpa,
+            comp_gpa=self.comp_gpa,
+            comp_227=self.comp_227,
+            preferred_sections=[
+                section.to_model() for section in self.preferred_sections
+            ],
+        )
+
+    def update(self, model: UTADetails, sections: list[SectionEntity]) -> None:
+        """
+        Update an ApplciationEntity from a UTA model.
+
+        Args:
+            model (UTA): The model to update the entity from.
+
+        Returns:
+            None
+        """
+        self.academic_hours = model.academic_hours
+        self.extracurriculars = model.extracurriculars
+        self.expected_graduation = model.expected_graduation
+        self.program_pursued = model.program_pursued
+        self.other_programs = model.other_programs
+        self.gpa = model.gpa
+        self.comp_gpa = model.comp_gpa
+        self.comp_227 = model.comp_227
+        self.preferred_sections = sections
+
+
+class New_UTA_Entity(UTAEntity):
```

Comment:
Recommend renaming this to `NewUTAApplicationEntity` - should remove `_` snake case syntax to better match Python conventions!